libraries/AP_HAL/utility/RingBuffer.cpp

```diff
@@ -47,8 +53,22 @@ void ByteBuffer::clear(void)
 
 uint32_t ByteBuffer::space(void) const
 {
+    if (size == 0) {
+        return 0;
+    }
+
+    /* use a copy on stack to avoid race conditions of @head being updated by
+     * the reader thread */
     uint32_t _head = head;
-    return size ? (_head > tail ? 0 : size) + _head - tail - 1 : 0;
```

Comment:
@OXINARF I'll send a separate PR for the coding style changes.... they are just removing the focus of this PR and I want to get this PR applied soon